src/cli.rs

```diff
@@ -261,21 +261,21 @@ impl Rover {
         Ok(git_context)
     }
 
-    pub(crate) fn get_reqwest_client(&self) -> RoverResult<Client> {
+    pub(crate) fn get_reqwest_client(&self) -> reqwest::Result<Client> {
```

Comment:
Should we continue to use RoverResult<Client> here? I'm wondering about the change in contract and whether it will mean the same thing. I was looking at [SputnikError](https://github.com/apollographql/rover/blob/f910779ae01496b68e23524e27022a027f3640f8/crates/sputnik/src/error.rs#L23), and it has a derived `From<reqwest::Error>` for `SputnikError::HttpError` with the description "HttpError occurs when an error occurs while reporting anonymous usage data.".
With that, I don't know if we want to assign problems with constructing the `reqwest` client to a `reqwest::Error` or assign it to an ad hoc error (via `anyhow`).

Seems like we also might want to consider swapping out `lazy-cell` in the future, as it hasn't been updated in a while and we may able to use tools in std for this (like `OnceLock`).

Reply:
The main thing was trying to pass _some_ error back to Sputnik from Rover, which can't be a `RoverError` because Sputnik can't depend on Rover.

As far as I know, though, Sputnik is _only_ used by Rover, so that whole trait boundary thing could be gotten rid of if we make this a `telemetry` module within Rover instead of a separate crate. I think that should be its own PR, though.

In general, I do avoid type-erasing errors with something like `anyhow` unless I need to, since it makes it harder to find out what's going on.